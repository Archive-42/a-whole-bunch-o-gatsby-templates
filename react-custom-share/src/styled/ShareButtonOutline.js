import { cx, css } from 'emotion';
import PropTypes from 'prop-types';
import React from 'react';

import colors from './colors';
import ShareButton from '../ShareButton';

const styles = css`
  align-items: center;
  background: none;
  border: 0;
  border-radius: 50%;
  cursor: pointer;
  display: flex;
  height: 44px;
  justify-content: center;
  margin: 3px;
  transition: 0.3s;
  width: 44px;

  & > svg {
    fill: white;
    filter: drop-shadow(0 0 2px rgba(0, 0, 0, 0.5));
    height: auto;
    transition: 0.3s;
    width: 28px;
  }

  @media (any-hover: none) {
    &:hover > svg {
      fill: white;
      filter: drop-shadow(0 0 2px rgba(0, 0, 0, 0.5));
    }
  }

  @media (min-width: 330px) {
    margin: 5px;
  }

  @media (min-width: 600px) {
    margin: 7px;
  }

  @media (min-width: 1025px) {
    margin: 10px;
  }
`;

const dynamicStyles = props => css`
  &:hover > svg {
    fill: ${colors[props.network]};
    filter: none;
  }
`;

const ShareButtonOutline = props => {
  const { className, ...rest } = props;
  const dynamicStylesStr = dynamicStyles(props);

  return (
    <ShareButton
      className={cx(styles, dynamicStylesStr, className)}
      {...rest}
    />
  );
};

ShareButtonOutline.propTypes = {
  className: PropTypes.string,
};

export default ShareButtonOutline;
