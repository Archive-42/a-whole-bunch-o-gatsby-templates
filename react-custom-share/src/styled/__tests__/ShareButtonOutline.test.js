import { createSerializer } from 'jest-emotion';
import { render, cleanup } from 'react-testing-library';
import * as emotion from 'emotion';
import FaTwitter from 'react-icons/lib/fa/twitter';
import React from 'react';

import ShareButtonOutline from '../ShareButtonOutline';

expect.addSnapshotSerializer(createSerializer(emotion));

afterEach(cleanup);

describe('<ShareButtonOutline>', () => {
  test(`renders unchanged`, () => {
    const props = { url: 'https://website-to-share.com', network: 'Twitter' };

    const { container } = render(
      <ShareButtonOutline {...props}>
        <FaTwitter />
      </ShareButtonOutline>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
