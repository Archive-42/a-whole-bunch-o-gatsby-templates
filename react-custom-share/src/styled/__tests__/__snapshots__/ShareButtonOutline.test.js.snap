// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShareButtonOutline> renders unchanged 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: 0;
  border-radius: 50%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 44px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 3px;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 44px;
}

.emotion-0 > svg {
  fill: white;
  -webkit-filter: drop-shadow(0 0 2px rgba(0,0,0,0.5));
  filter: drop-shadow(0 0 2px rgba(0,0,0,0.5));
  height: auto;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 28px;
}

@media (any-hover:none) {
  .emotion-0:hover > svg {
    fill: white;
    -webkit-filter: drop-shadow(0 0 2px rgba(0,0,0,0.5));
    filter: drop-shadow(0 0 2px rgba(0,0,0,0.5));
  }
}

@media (min-width:330px) {
  .emotion-0 {
    margin: 5px;
  }
}

@media (min-width:600px) {
  .emotion-0 {
    margin: 7px;
  }
}

@media (min-width:1025px) {
  .emotion-0 {
    margin: 10px;
  }
}

.emotion-0:hover > svg {
  fill: rgba(99,205,241,1);
  -webkit-filter: none;
  filter: none;
}

<button
  aria-label="Share on Twitter"
  class="emotion-0 rcs-button"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align: middle;"
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m37.7 9.1q-1.5 2.2-3.7 3.7 0.1 0.3 0.1 1 0 2.9-0.9 5.8t-2.6 5.5-4.1 4.7-5.7 3.3-7.2 1.2q-6.1 0-11.1-3.3 0.8 0.1 1.7 0.1 5 0 9-3-2.4-0.1-4.2-1.5t-2.6-3.5q0.8 0.1 1.4 0.1 1 0 1.9-0.3-2.5-0.5-4.1-2.5t-1.7-4.6v0q1.5 0.8 3.3 0.9-1.5-1-2.4-2.6t-0.8-3.4q0-2 0.9-3.7 2.7 3.4 6.6 5.4t8.3 2.2q-0.2-0.9-0.2-1.7 0-3 2.1-5.1t5.1-2.1q3.2 0 5.3 2.3 2.4-0.5 4.6-1.7-0.8 2.5-3.2 3.9 2.1-0.2 4.2-1.1z"
      />
    </g>
  </svg>
</button>
`;
